src/test/java/com/example/hashcartapp/AdvertisementServiceTests.java

```diff
@@ -0,0 +1,73 @@
+package com.example.hashcartapp;
+
+import com.example.hashcartapp.entities.Advertisement;
+import com.example.hashcartapp.repository.AdvertisementRepository;
+import com.example.hashcartapp.service.AdvertisementService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Pageable;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class AdvertisementServiceTests {
+
+    @Mock
+    AdvertisementRepository advertisementRepository;
+
+    @InjectMocks
+    AdvertisementService advertisementService;
+
+    @Test
+    public void given_get_all_advertisements_should_return_list_of_all_advertisements(){
+        advertisementService.getAllAdvertisements();
+        verify(advertisementRepository,times(1)).findAll();
```

Comment:
testing could be made better with better assertions and using mockito's when- thenReturn apis. Also try to cover negative scenarios.
some references - 
https://www.baeldung.com/mockito-annotations
https://www.springboottutorial.com/spring-boot-unit-testing-and-mocking-with-mockito-and-junit